src/indexer/merger.rs

```diff
@@ -1256,6 +1256,7 @@ mod tests {
     }
 
     #[test]
+    #[should_panic(expected = "Segment_ids cannot be empty.")]
```

Comment:
I'm having trouble implementing the last two stages,
since IndexWriter#wait_merging_threads moves index_writer,
and I cannot explicitly trigger a commit.
I am still looking for better ways to implement this,
although any help or advice would be appreciated.

Reply:
Which commit is this? On the head, the test still is annotated "should_panic".

> After merge, the documents are deleted in the new segment, even before the commit is executed.
Is this expected behaviour?

Probably yes, but I would like to see the actual code/behavior to be sure.
With this order, I would expect:
- before the call to commit, and after merge has succeeded, the number of documents should be unchanged. The segments being merged are still here.
- after the call to commit, they should be replaced by no segments at all.

Reply:
> I'm having trouble implementing the last two stages,
> since IndexWriter#wait_merging_threads moves index_writer,
> and I cannot explicitly trigger a commit.

call `.wait_merging_thread` after the ast commit.

Reply:
> > I'm having trouble implementing the last two stages,
> > since IndexWriter#wait_merging_threads moves index_writer,
> > and I cannot explicitly trigger a commit.
> 
> calling `.wait_merging_thread` after the ast commit should be ok. Also `start_merger` itself returns a future you can wait on.

https://github.com/tantivy-search/tantivy/pull/472/commits/ba45c6e50ebcdb73c646a8da161209c50678cc31

Reply:
> before the call to commit, and after merge has succeeded, the number of documents should be unchanged. The segments being merged are still here.

As you can probably see in the latest commit, after calling merge,
no documents are returned by the searcher.

Reply:
Can you call `.load_searcher()` before you acquire your searcher to test the number of docs?

Reply:
I just pushed another commit,
Seems like there are still 0 docs though :(.

Reply:
I did.
I also added a test with comments, which now fails.
If I understood correctly, documents should still exist after the merge and before the commit.
Unfortunately, this does not seem to be the case in the latest [commit](https://github.com/tantivy-search/tantivy/pull/472/commits/65d88677a8cdae68d1fba3239a29ce2dd3b0271a).

Reply:
@barrotsteindev I think you found a bug. It's there in master too. I will hotfix that in 0.8. Sorry I need to take of my daughter right now.

Reply:
@barrotsteindev Created an issue. This is terrible but thank you so much for spotting that!